lib/Sema/CSBindings.cpp

```diff
@@ -547,13 +547,25 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) const {
       }
       break;
     }
+    case ConstraintKind::KeyPathApplication: {
+      if (result.FullyBound)
+        continue;
+
+      // If this variable is in the application projected result type, it is
+      // fully bound.
```

Comment:
Could you add a bit to this comment saying something like:

```
mark the result as FullyBound to ensure we delay binding until
we've bound other type variables. In particular, this ensures we
try to bind the key path type first, which can allow us to 
discover additional bindings for the result type. 
```

Just to clarify the use of `FullyBound`, which IMO has a pretty confusing name.

Reply:
Sure, I agree... TBH I didn't know the concept, had to look on the comments around the code to understand what this means and how the bindings work.

Reply:
Yeah the current binding logic is pretty confusing, we're hoping to refactor it to be simpler and more efficient :)

Reply:
Just added! :)

Reply:
Awesome, thank you!